routes.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+const Joi = require('joi');
+let internals = {};
+internals.getUsers = function (request, reply) {
+  if (request.query.name) {
+    return reply(internals.findUsers(request.query.name));
+  }
+  reply(internals.users);
+}
+
+internals.findUsers = function (name) {
+  return internals.users.filter((product) => {
+    return product.name.toLowerCase() === name.toLowerCase();
+  });
+}
+
+internals.getProduct = function (request, reply) {
+  const filtered = internals.users.filter((product) => {
+    return product.id === parseInt(request.params.id);
+  }).pop();
+  reply(filtered);
+}
+
+internals.addProduct = function (request, reply) {
+  const product = {
+    id: internals.users[internals.users.length - 1].id + 1
+    , name: request.payload.name
+  };
+  internals.users.push(product);
+  reply(product).created('/users/' + product.id);
+}
+
+module.exports = [{
+  method: 'GET',
+  path: '/users',
+  config: {
+    validate: {
+      query: {
+        name: Joi.string()
+      }
+    },
+    handler: internals.getUsers
+  }
+}, {
+  method: 'GET',
+  path: '/users/{id}',
+  handler: internals.getProduct
+}, {
```

Comment:
lets remove POST route. we have another issue for doing it. Lets do it there. actually i thought of doing initailize routes there.